src/modules/roc_netio/target_uv/roc_netio/transceiver.cpp

```diff
@@ -185,14 +158,32 @@ void Transceiver::remove_port(packet::Address bind_address) {
     if (!task.result) {
         roc_panic("transceiver: can't remove port %s: unknown port",
                   packet::address_to_str(bind_address).c_str());
+    } else {
+        roc_panic_if_not(task.port);
+        wait_port_removed_(*task.port);
     }
 }
 
-void Transceiver::run() {
-    if (!valid()) {
-        roc_panic("transceiver: can't use invalid transceiver");
+void Transceiver::handle_closed(BasicPort& port) {
+    roc_log(LogDebug, "transceiver: asynchronous close finished: port %s",
+            packet::address_to_str(port.address()).c_str());
+
+    core::Mutex::Lock lock(mutex_);
+
+    for (core::SharedPtr<BasicPort> pp = ports_.front(); pp; pp = ports_.nextof(*pp)) {
+        if (pp.get() == &port) {
+            ports_.remove(*pp);
+            cond_.broadcast();
+
+            return;
+        }
     }
 
+    roc_panic("transceiver: can't remove unknown port %s",
+              packet::address_to_str(port.address()).c_str());
```

Comment:
Since all this stuff with multiple async_close() and handle_closed() calls is quite complicated, I would remove this panic and just allow multiple handle_closed() calls.

Reply:
Fixed

Reply:
Fixed